text/0132-indexes-api.md

```diff
@@ -0,0 +1,217 @@
+# Indexes API
+
+## 1. Summary
+
+This specification describes the indexes API endpoints permitting to list, create, update and delete Meilsearch indexes.
+
+## 2. Motivation
+N/A
+
+## 3. Functional Specification
+
+Indexes contain a set of documents to be searchable and have their specific settings.
+
+See [Documents API specification](0124-documents-api.md) and [Settings API specification](0123-settings-api.md) for details.
+
+### 3.1. API Endpoints Definition
+
+
+### 3.1. `index` API Resource Definition
+
+| Field                         | Type            | Required |
+|-------------------------------|-----------------|----------|
+| [uid](#311-uid)               | String          | True     |
+| [name](#312-name)             | String          | False    |
+| [primaryKey](#313-primaryKey) | String / `null` | False    |
+| [createdAt](#314-createdAt)   | String          | False    |
+| [updatedAt](#315-updatedAt)   | String          | False    |
+
+#### 3.1.1. `uid`
+
+- Type: String
+- Required: True
+
+A unique identifier for the index.
+
+This field is mandatory when creating an index and cannot be changed afterwards.
+
+#### 3.1.2. `name`
+
+- Type: String
+- Required: False
+
+This field is automatically generated by the engine when creating an index and is equal to the value of `uid`.
+
+It is not possible to fill it in when creating or updating an index.
+
+#### 3.1.3. `primaryKey`
+
+- Type: String
+- Required: False
+- Default: `null`
+
+Allows to bypass the auto-inference mechanism of the document identifiers.
+
+By default, the `primaryKey` will be chosen by the auto-inference mechanism by the engine when a first document is indexed.
+
+Specifying this field tells the engine to use the document attribute specified in `primaryKey` and bypasses this mechanism.
+
+When the index is empty, it is possible to modify the `primaryKey`.
```

Comment:
I think it is important yes since I'm always answering to users asking how to change their primary key that they `cannot change it whenever they set it or once they added one document`. 

I think the doc should be updated on the matter

Reply:
@eskombro is maybe also not aware this behavior changed. It may (or not in which case sorry) be of interest to him.

Reply:
Documentation issue created: https://github.com/meilisearch/documentation/issues/1574